include/click/standard/storage.hh

```diff
@@ -21,6 +24,17 @@ class Storage { public:
     index_type head() const		{ return _head; }
     index_type tail() const		{ return _tail; }
 
+    index_type take_tail() {
+        index_type shadow_tail;
+        index_type next_shadow_tail;
+        do {
+            shadow_tail = _tail;
+            next_shadow_tail = next_i(shadow_tail);
+            if (next_shadow_tail == _head) return index_error;
+        } while (shadow_tail != reinterpret_cast<atomic_uint32_t*>(const_cast<index_type*>(&_tail))->compare_swap(shadow_tail,next_shadow_tail));
```

Comment:
Prefer this line:

```
} while (shadow_tail != atomic_uint32_t::compare_swap(_tail, shadow_tail, next_shadow_tail));
```

i.e. use the static version of atomic_uint32_t::compare_swap.